### PR TITLE
fixes #1259 now ignores protocol of Dockerfile EXPOSE

### LINKED ISSUE #1259: EXPOSE parser doesn't understand port protocol
something like `EXPOSE 8888/tcp` is not supported, if you are using some else's image that was not tailored for dokku this can easily happen.

### PR DESCRIPTION
**Pull request by kilianc**
updating tests now

### CODE REVIEW

File: plugins/00_dokku-standard/commands [hunk 1]
```diff
@@ -36,7 +36,7 @@ case "$1" in
 
       dockerfile)
         # extract first port from Dockerfile
-        DOCKERFILE_PORT=$(grep EXPOSE Dockerfile | head -1 | awk '{ print $2 }' || true)
+        DOCKERFILE_PORT=$(grep EXPOSE Dockerfile | head -1 | awk '{ print $2 }' | sed 's/\/.*//' || true)
```
Comment (josegonzalez) [inline plugins/00_dokku-standard/commands]: Should we ensure this is a tcp port and not a udp one?

Comment (kilianc) [inline plugins/00_dokku-standard/commands]: I managed to do that let's wait for the test suite result

**Comment by kilianc** [comment]
btw, I feel like we're doing it wrong, we should probably inspect the container and look for first open tcp port

**Comment by josegonzalez** [comment]
The container isn't running yet though, is it? And I think we need this information beforehand (though maybe I'm wrong).

**Comment by kilianc** [comment]
@josegonzalez this is how it works now, but it's doable because we turn it on -> grab port -> wait for checks -> reload nginx -> kill old container

It may require some work, so for the time being I would just merge this if it works and iterate again

**Comment by josegonzalez** [comment]
And that regex works if you don't use `/tcp`, right?

**Comment by kilianc** [comment]
:+1:

**Comment by josegonzalez** [comment]
Seems like tests fail.

**Comment by kilianc** [comment]
Seems like it, I will double check in the next days

**Comment by kilianc** [comment]
@josegonzalez it appears that grep has different regexp implementations in different platforms, so I rewrote it using basic POSIX. it's ugly but it works.

Any advice on how to test both `EXPOSE 3000` and `EXPOSE 3000/tcp` ? I don't want to have 2 docker apps and the test suite is insanely slow already... I wish we had some unit testing in place for this

**Comment by josegonzalez** [comment]
@michaelshobbs given that you implemented dockerfile support, thoughts on his last question?

**Comment by michaelshobbs** [comment]
eh well, if the point of the test is to have a running dockerfile container with two different formats, I'm not sure we have a good way right now. Stubbing out the deployment functions for this type of stuff that doesn't actually require a `curl`-style request at the end would be nice.

**Comment by kilianc** [comment]
> EDIT: maybe make it a function, source the functions file in the test and then use bats from there?

yep this is what I was looking for, I will add a `get_exposed_port()` in this PR

**Comment by michaelshobbs** [comment]
how's this looking @kilianc?

**Comment by kilianc** [comment]
@michaelshobbs no time this week, bee busy on another project :/ feel free to take over

**Comment by josegonzalez** [comment]
@michaelshobbs want to take a crack at this? Just this and the other pr before I can make a release.

**Comment by michaelshobbs** [comment]
@josegonzalez yes. #1311

**Comment by josegonzalez** [comment]
Closing as #1311 is now merged. Thanks @kilianc for reporting the issue and providing an initial fix :)

**Pull request merged**